src/base/misc/fatfs.c

```diff
@@ -1585,7 +1585,13 @@ void build_boot_blk(fatfs_t *f, unsigned char *b)
     case OLDMSD_D:
       txfr->ofs = 0x0000;
       txfr->seg = 0x0070;
-      txfr->ax  = d_o >> 16;
+      /*
+       * MS-DOS 3.10 needs the offset of MSDOS.SYS / 16 passed in AX and other
+       * versions don't seem to mind. See the issue discussion at
+       * https://github.com/stsp/dosemu2/issues/278. Note this relies upon the
+       * ordering of the directory, as was a requirement with earlier DOSes.
```

Comment:
Why do you think so?
The original calcs do require ordering, but
ours are not: we take the offset of msdos.sys directly,
so it can probably be anywhere?
Please also update the patch description with this info.